sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs

```diff
@@ -209,6 +210,75 @@ public async Task EventHub_PartitionKey()
             }
         }
 
+        [Test]
+        public async Task EventHub_InitialOffsetFromStart()
+        {
+            using (var host = BuildHost<EventHubTestSendOnlyJobs>().Item1)
+            {
+                var method = typeof(EventHubTestSendOnlyJobs).GetMethod(nameof(EventHubTestSendOnlyJobs.SendEvent_TestHub), BindingFlags.Static | BindingFlags.Public);
+                await host.CallAsync(method, new { input = _testId });
+            }
+            var initialOffsetOptions = new InitialOffsetOptions()
+            {
+                Type = "FromStart"
+            };
+            using (var host = BuildHost<EventHubTestInitialOffsetFromEndJobs>(initialOffsetOptions: initialOffsetOptions).Item1)
+            {
+                bool result = _eventWait.WaitOne(Timeout);
+                Assert.True(result);
+            }
+        }
+
+        [Test]
+        public async Task EventHub_InitialOffsetFromEnd()
+        {
+            // Send a message to ensure the stream is not empty as we are trying to validate that no messages are delivered in this case
+            using (var host = BuildHost<EventHubTestSendOnlyJobs>().Item1)
+            {
+                var method = typeof(EventHubTestSendOnlyJobs).GetMethod(nameof(EventHubTestSendOnlyJobs.SendEvent_TestHub), BindingFlags.Static | BindingFlags.Public);
+                await host.CallAsync(method, new { input = _testId });
+            }
+            var initialOffsetOptions = new InitialOffsetOptions()
+            {
+                Type = "FromEnd"
+            };
+            using (var host = BuildHost<EventHubTestInitialOffsetFromEndJobs>(initialOffsetOptions: initialOffsetOptions).Item1)
+            {
+                // We don't expect to get signalled as there should be no messages received with a FromEnd initial offset
+                bool result = _eventWait.WaitOne(Timeout);
+                Assert.False(result, "An event was received while none were expected.");
+            }
+        }
+
+        [Test]
+        public async Task EventHub_InitialOffsetFromEnqueuedTime()
+        {
+            // Mark the time now and send a message which should be the only one that is picked up when we run the actual test host
+            DateTime initialOffsetEnqueuedTimeUTC = DateTime.UtcNow;
```

Comment:
Considering there are no other messages in the hub when the test starts how would this ever fail?

Reply:
Well it would fail if it filtered out the one event that was sent, but yes it is not verifying the opposite case - namely that it will correctly filter out something that is earlier.

Reply:
You can remove the entire feature being added in this PR and this test would keep passing.

Reply:
I've updated it so that it filters out messages that are earlier.